tests/integration/test_fim/test_files/test_audit/test_audit_multiple_keys.py

```diff
@@ -0,0 +1,112 @@
+# Copyright (C) 2015-2021, Wazuh Inc.
+# Created by Wazuh, Inc. <info@wazuh.com>.
+# This program is free software; you can redistribute it and/or modify it under the terms of GPLv2
+import os
+
+import pytest
+import wazuh_testing.fim as fim
+
+from wazuh_testing import global_parameters
+from wazuh_testing.tools import PREFIX
+from wazuh_testing.tools.configuration import load_wazuh_configurations, check_apply_test
+
+# Marks
+
+pytestmark = [pytest.mark.linux, pytest.mark.tier(level=1)]
+
+
+local_internal_options = {'syscheck.debug': '2', 'analysisd.debug': '2', 'monitord.rotate_log': '0'}
+daemons_handler_configuration = {'daemons': ['wazuh-syscheckd', 'wazuh-analysisd', 'wazuh-modulesd']}
+
+
+
+# Variables
+
+monitored_test_dir = os.path.join(PREFIX, 'monitored_test_dir')
+non_monitored_test_dir = os.path.join(PREFIX, 'non_monitored_test_dir')
+custom_keys = ['key1', 'key2']
+param_list = []
+audit_rules_reload_interval = 40
+
+param_list = ' '.join([f"-k {key}" for key in custom_keys])
+
+test_directories = [monitored_test_dir, non_monitored_test_dir]
+test_data_path = os.path.join(os.path.dirname(os.path.realpath(__file__)), 'data')
+configurations_path = os.path.join(test_data_path, 'wazuh_conf_multiple_audit_keys.yaml')
+
+# Configurations
+
+config_params, config_metadata = fim.generate_params(extra_params={'MONITORED_DIR': monitored_test_dir,
+                                                     'AUDIT_KEYS': ','.join(key for key in custom_keys)},
+                                                     modes=['whodata'])
+
+configurations = load_wazuh_configurations(configurations_path, __name__,
+                                           params=config_params, metadata=config_metadata)
+
+
+# fixtures
+
+@pytest.fixture(scope='module', params=configurations)
+def get_configuration(request):
+    """Get configurations from the module."""
+    return request.param
+
+
+@pytest.fixture(scope='module', params=configurations)
+def set_audit_rules():
+    # Create the custom audit rules for the non monitored directory
+    fim.run_audit_command(directory=non_monitored_test_dir, params=param_list, cmd_type='add')
+    # Remove audit rule that FIM configures for each monitored directory
+    fim.run_audit_command(directory=monitored_test_dir, params='-k wazuh_fim', cmd_type='delete')
+    # Set the audit rule for the monitored directory with more than one key
+    fim.run_audit_command(directory=monitored_test_dir, params='-k wazuh_fim -k "a_random_key"', cmd_type='add')
+
+    yield
+
+    # Remove the audit rules configured by the test
+    fim.run_audit_command(directory=non_monitored_test_dir, params=param_list, cmd_type='delete')
+    fim.run_audit_command(directory=monitored_test_dir, params=param_list, cmd_type='delete')
```

Comment:
This change should fix this issue https://github.com/wazuh/wazuh-qa/pull/1192#issuecomment-920754399

```suggestion
@pytest.fixture(scope='function')
def set_audit_rules():
    # Create the custom audit rules for the non monitored directory
    fim.run_audit_command(directory=non_monitored_test_dir, params=param_list, cmd_type='add')
    # Remove audit rule that FIM configures for each monitored directory
    fim.run_audit_command(directory=monitored_test_dir, params='-k wazuh_fim', cmd_type='delete')
    # Set the audit rule for the monitored directory with more than one key
    fim.run_audit_command(directory=monitored_test_dir, params='-k wazuh_fim -k "a_random_key"', cmd_type='add')

    yield

    # Remove the audit rules configured by the test
    fim.run_audit_command(directory=non_monitored_test_dir, params=param_list, cmd_type='delete')
    fim.run_audit_command(directory=monitored_test_dir, params=param_list, cmd_type='delete')
```